R/FilteredData.R

```diff
@@ -150,11 +152,35 @@ FilteredData <- R6::R6Class( # nolint
       checkmate::assert_class(x, "reactive")
       private$available_teal_slices <- reactive({
         # Available filters should be limited to the ones relevant for this FilteredData.
-        Filter(function(x) x$dataname %in% self$datanames(), x())
+        allowed <- attr(self$get_filter_state(), "include_varnames")
+        forbidden <- attr(self$get_filter_state(), "exclude_varnames")
```

Comment:
We want this reactive to be dependent on `x` (slices_global)

```suggestion
        allowed <- attr(isolate(self$get_filter_state()), "include_varnames")
        forbidden <- attr(isolate(self$get_filter_state()), "exclude_varnames")
```